libs/variant/include/variant/variant.hpp

```diff
@@ -598,12 +598,12 @@ struct ForwardSerializer<fetch::variant::Variant, D>
     {
     case Type::Type::UNDEFINED:
     {
-      // TODO(tfr): Work out if msgpack has support for undefined
-      return;
+      throw std::runtime_error{"Unsupported variant type for serialisation"};
     }
     case Type::Type::NULL_VALUE:
     {
-      // TODO(tfr): Work out if msgpack has support for null
+      serializer.Allocate(sizeof(uint8_t));
+      serializer.WriteByte(static_cast<uint8_t>(TypeCodes::NIL));
```

Comment:
I think it would be preferable if this could be made less manual, i.e.
```serializer << nullptr;```

Reply:
Done